src/test/test-specifier.c

```diff
@@ -138,4 +138,18 @@ TEST(specifiers) {
         }
 }
 
+TEST(specifiers_missing_data_ok) {
+        _cleanup_free_ char *resolved = NULL;
+
+        assert_se(setenv("SYSTEMD_OS_RELEASE", "/dev/null", 1) == 0);
+        assert_se(specifier_printf("%A-%B-%M-%o-%w-%W", SIZE_MAX, specifier_table, NULL, NULL, &resolved) >= 0);
+        assert_se(streq(resolved, "-----"));
+
+        assert_se(setenv("SYSTEMD_OS_RELEASE", "/nosuchfileordirectory", 1) == 0);
+        assert_se(specifier_printf("%A-%B-%M-%o-%w-%W", SIZE_MAX, specifier_table, NULL, NULL, &resolved) == -ENOENT);
```

Comment:
maybe this behaviour isn't ideal. i.e. maybe we should treat a missing os-release file like an empty one in this context,

Reply:
I thought about this, and that was my initial thought. But we generally expect os-release to be there. And if you're calling 'enable' on files with specifiers, I think it's more likely that something went wrong with the installation and os-release is missing (or in the wrong path or something) than that it's missing on purpose. We only throw the error if specifiers that refer to os-release are used. And it is OK to have an empty os-release file, that is tested in one of the tests. This also lightly pushes people towards installing os-release always, which I think is a reasonable thing to do.